pkg/tfshim/shim.go

```diff
@@ -244,6 +244,7 @@ type Provider interface {
 
 	// Checks if a value is representing a Set, and unpacks its elements on success.
 	IsSet(ctx context.Context, v interface{}) ([]interface{}, bool)
+	DetailedSchemaDump() []byte
```

Comment:
Let's make sure this has an explanation (and that we give it a `context.Contex` and an error channel. I would like us to avoid communicating errors with `panic`.
```suggestion
	// DetailedSchemaDump returns output useful for provider authors to diagnose 
	// bugs in the bridge. It does not have a stable format and should not be
	// parsed.
	//
	// If a provider does not support dumping it's raw schema, it should return
	// (nil, nil).
	DetailedSchemaDump(context.Context) ([]byte, error)
```